opossum-tool/OpossumUI
# Split linter and formatter

### Summary of changes

- introduce scripts for checking/writing with prettier and run on entire repo instead of just src files
- do not produce linter errors on formatting issues
- fix formatting issues as part of pre-commit hook
- validate formatting in CI/CD
- ignore certain root files from linting

### Context and reason for change

Fixes the following issues:
- IDE unnecessarily highlighting auto-fixable formatting issues as errors
- not all repo files are auto-formatted
- some root files were showing linter-config inclusion/exclusion errors

### How can the changes be tested

1. Locally test linter and formatter commands (`yarn lint` + `yarn format`).
2. See CI/CD run of formatting check

Commit-wise review recommended.
